src/mjolnir/graphbuilder.cc

```diff
@@ -715,7 +715,8 @@ void BuildTileSet(const std::string& ways_file,
           // Set use to ramp or turn channel
           if (edge.attributes.turn_channel) {
             directededge.set_use(Use::kTurnChannel);
-          } else if (edge.attributes.link) {
+            // Do not overwrite rest area or service area use for ramps
+          } else if (edge.attributes.link && (use != Use::kServiceArea && use != Use::kRestArea)) {
```

Comment:
@gknisely I made this change to allow links to be marked with rest or service area use, previously a motorway_link was getting ramp use instead of rest/service use. Can you take a look, is this the right thing to do here?

Reply:
seems correct.  @kdiluca can you run this on a small extract.